pkg/flags/flags.go

```diff
@@ -264,6 +280,10 @@ func ParseArgs(args []string) (*EnvFlags, error) {
 		panic(fmt.Errorf("--fail-fast and --parallel are mutually exclusive options"))
 	}
 
+	if kubeconfig == "" && kubeContext != "" {
+		panic(fmt.Errorf("--kube-context requires the --kubeconfig option"))
```

Comment:
While this is true, it is not necessary to have the --kubeconfig flag for this to work. This is because if [`ResolveKubeConfigFile`](https://github.com/kubernetes-sigs/e2e-framework/blob/db4e9fe7e23d2602c34d1fe3853625e754b1fa41/klient/conf/config.go#L67) is called during the construction of the client instance, it will attempt to resolve the kubeconfig automatically.  

Maybe that check should be moved [above](https://github.com/kubernetes-sigs/e2e-framework/pull/197/files#diff-3923bb7d5d40d33c81ef1c40584ce3ef3f81c7c6bb1ffb2adfc842b745533a7cR41) when setting the flags up.